src/antispoof.c

```diff
@@ -64,23 +72,77 @@ struct pg_antispoof_config {
 /* Size of the "Sender" part of struct pg_antispoof_arp */
 #define PG_ANTISPOOF_ARP_CHECK_PART2 (6 + 4)
 
-void pg_antispoof_arp_enable(struct pg_brick *brick, uint32_t ip)
+void pg_antispoof_arp_enable(struct pg_brick *brick)
 {
-	struct pg_antispoof_state *state;
+	pg_brick_get_state(brick, struct pg_antispoof_state)->arp_enabled =
+		true;
+}
 
-	state = pg_brick_get_state(brick, struct pg_antispoof_state);
-	state->ip = ip;
+int pg_antispoof_arp_add(struct pg_brick *brick, uint32_t ip,
+			 struct pg_error **errp)
+{
+	struct pg_antispoof_state *state =
+		pg_brick_get_state(brick, struct pg_antispoof_state);
+	uint16_t n = state->arp_size;
+
+	if (n == ARP_MAX) {
+		*errp = pg_error_new("Maximal IP reached");
+		return -1;
+	}
+	struct arp *arp = &state->arp[n];
```

Comment:
you can do this before the check, as long as you don't access arp, you should not get any segfault